internal/processor/stream/processor.go

```diff
@@ -260,13 +283,30 @@ func (p *Processor) HandleStream(
 			}
 			result.AddAccepted(len(batch))
 		}
-		if readErr == io.EOF {
-			break
-		} else if readErr != nil {
-			return readErr
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				return nil
+			}
+			return err
+		}
+	}
+}
+
+func (p *Processor) asyncProcessor(
+	ctx context.Context,
+	processor model.BatchProcessor,
+	batchChan chan *model.Batch,
+	result *Result,
```

Comment:
`result` isn't updated - should it even be passed in? I think it should only be updated by the synchronous code path, since it'll be returned to the client.